react/features/rtcstats/RTCStats.ts

```diff
@@ -185,6 +198,38 @@ class RTCStats {
         this.trace?.close();
     }
 
+    /**
+     * RTCStats client can notify the APP of any PeerConnection related event that occurs.
+     *
+     * @param {Object} event - Rtcstats event.
+     * @returns {void}
+     */
+    handleRtcstatsEvent(event: any) {
+        switch (event.type) {
+        case PC_CON_STATE_CHANGE: {
+            const { body: { isP2P, state } = { state: null,
+                isP2P: null } } = event;
+
+            this.connStateEvents.push(event.body);
+
+            // We only report PC related connection issues. If the rtcstats websocket is not connected at this point
+            // it usually means that none of our services can be reached i.e. there's problem with the internet
+            // connection and not necessarily with reaching the JVB (due to a firewall or other reasons).
+            if (state === PC_STATE_FAILED && this.trace.isConnected()) {
+                const connectionType = isP2P ? 'P2P' : 'JVB';
+                const wasConnected = this.connStateEvents.some((connectionEvent: { isP2P: any; state: string; }) =>
+                    (connectionEvent.isP2P === isP2P) && (connectionEvent.state === PC_STATE_CONNECTED));
+
+                logger.info(`${connectionType} PeerConnection failed, previously connected: ${wasConnected}`);
+
+                APP.API.notifyPeerConnectionFailure(isP2P, wasConnected);
```

Comment:
You need to check if typeof APP is undefined before calling this or mobile will die.

Reply:
ow, we don't want that, fixed.